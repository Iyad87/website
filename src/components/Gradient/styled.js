import styled from 'styled-components';

export const BaseGradient = styled.div`
    position: absolute;
    bottom: 0;
    left: -7.5%;
    right: -7.5%;
    height: ${props => props.height};
    background-image: linear-gradient(
      -180deg,
      rgba(255, 255, 255, 0.5) 0%,
      #f3f5fa 52%
    );
    z-index: -1;
  }
`;
