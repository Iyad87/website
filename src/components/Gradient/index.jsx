import React from 'react';
import PropTypes from 'prop-types';
import { BaseGradient } from './styled';

const Gradient = ({ height, ...props }) => (
  <BaseGradient height={height} {...props} />
);

Gradient.propTypes = {
  height: PropTypes.string,
};

Gradient.defaultProps = {
  height: '50%',
};

export default Gradient;
