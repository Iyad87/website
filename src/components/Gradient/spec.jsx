import React from 'react';
import { shallow } from 'enzyme';
import Gradient from './index';

test('snapshot', () => {
  const wrapper = shallow(<Gradient />);

  expect(wrapper).toMatchSnapshot();
});

test('applies height property', () => {
  const wrapper = shallow(<Gradient height="100px" />);

  expect(wrapper.prop('height')).toEqual('100px');
});
