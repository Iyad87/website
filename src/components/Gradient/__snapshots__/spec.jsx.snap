// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<styled__BaseGradient
  height="50%"
/>
`;
